scio-core/src/main/scala/com/spotify/scio/util/Functions.scala

```diff
@@ -74,7 +74,7 @@ private[scio] object Functions {
       accumulator._2.asScala.foldLeft(accumulator._1)(s)
 
     override def mergeAccumulators(accumulators: JIterable[(U, JList[T])]): (U, JList[T]) = {
-      val combined = accumulators.asScala.map(fold).reduce(c)
+      val combined = accumulators.asScala.iterator.map(fold).reduce(c)
```

Comment:
we can get `AbstractIterator` as well by getting the `iterator` from the java collection, i.e `iterator.asScala`. (which is also the strategy that flink uses). I know that use cases may vary ... but i would probably make: `before converting to scala get the iterator` a rule of thumb.